packages/sdk/src/addresses/index.ts

```diff
@@ -119,25 +119,31 @@ export async function getAddresses({
 export function getAccountDataFromHdNode({
   hdNode,
   format = "legacy",
-  network = "testnet"
+  network = "testnet",
+  accountIndex = 0,
+  index = 0
 }: GetAccountDataFromHdNodeOptions) {
   if (!hdNode) {
     throw new Error("Invalid options provided.");
   }
 
   const addressType = addressNameToType[format];
-  //
-  const child = hdNodeToChild(hdNode, format, 0);
+
+  const fullDerivationPath = getDerivationPath(format, accountIndex, index)
+  const child = hdNode.derivePath(fullDerivationPath)
+
   const pubKey = format === "taproot" ? toXOnly(child.publicKey) : child.publicKey;
   const paymentObj = createTransaction(pubKey, addressType, network);
 
   const address = paymentObj.address!;
+
   const account: Account = {
     address,
     pub: child.publicKey.toString("hex"),
     priv: child.privateKey!.toString("hex"),
     format,
-    type: addressType
+    type: addressType,
+    derivationPath: fullDerivationPath
```

Comment:
right now, there is no specific use case yet, but I am adding this to have better visibility to each addresses in the addressList. 
It can potentially be used to solve issues I mentioned above like knowing the last index to generate address from.